packages/engine/paima-funnel/src/funnels/parallelEvm/funnel.ts

```diff
@@ -85,7 +85,10 @@ export class ParallelEvmFunnel extends BaseFunnel implements ChainFunnel {
       );
 
       if (queryResults[0]) {
-        cachedState.lastBlock = queryResults[0].block_height;
+        cachedState.lastBlock = Math.max(
```

Comment:
It's fine because this is done inside `readData`, and that function is not called if the presync is not finished. And this cache entry is not used to decide that.